solidity/test/integration/Slashing.test.ts

```diff
@@ -0,0 +1,531 @@
+
+          expect(await staking.getSlashingQueueLength()).to.equal(
+            walletMembers.length
+          )
+
+          for (let i = 0; i < walletMembers.length; i++) {
+            const slashing = await staking.slashingQueue(i)
+
+            expect(slashing.amount).to.equal(
+              amountToSlash,
+              `unexpected slashing amount for ${i}`
+            )
+
+            expect(slashing.stakingProvider).to.equal(
+              walletMembers[i].stakingProvider,
+              `unexpected staking provider for ${i}`
+            )
+          }
+        })
+
+        it("should close the wallet in the wallet registry", async () => {
+          // eslint-disable-next-line @typescript-eslint/no-unused-expressions
+          expect(await walletRegistry.isWalletRegistered(ecdsaWalletID)).to.be
+            .false
+        })
+
+        it("should close the wallet in the bridge", async () => {
+          const storedWallet = await bridge.wallets(walletPubKeyHash160)
+
+          expect(storedWallet.state).to.be.equal(walletState.Terminated)
+        })
+
+        it("should consume around 3 100 000 gas for Bridge.notifyMovingFundsTimeoutTx transaction", async () => {
+          await assertGasUsed(
+            notifyFraudChallengeDefeatTimeoutTx,
+            3_150_000,
+            100_000
+          )
+        })
+      })
+    })
+  })
+
+  describe("notifyRedemptionTimeout", async () => {
+    before(async () => {
+      await createSnapshot()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    describe("when wallet is created", async () => {
+      const deposit = SingleP2SHDeposit.deposits[0]
+
+      const { walletPubKeyHash: walletPubKeyHash160 } = deposit.reveal
+      const { walletPublicKey, walletID: ecdsaWalletID } = deposit.ecdsaWallet
+
+      let walletMembers: Operators
+      let redeemerOutputScript: BytesLike
+
+      before(async () => {
+        const requestNewWalletTx = await bridge.requestNewWallet(NO_MAIN_UTXO)
+
+        await produceRelayEntry(walletRegistry, randomBeacon)
+        ;({ walletMembers } = await performEcdsaDkg(
+          walletRegistry,
+          walletPublicKey,
+          requestNewWalletTx.blockNumber
+        ))
+
+        const { fundingTx, reveal } = SingleP2SHDeposit.deposits[0]
+        reveal.vault = tbtcVault.address
+
+        // We use a deposit funding bitcoin transaction with a very low amount,
+        // so we need to update the dust and redemption thresholds to be below it.
+        await updateDepositDustThreshold(10_000) // 0.0001 BTC
+        await updateRedemptionDustThreshold(2_000) // 0.00002 BTC
+
+        await bridge.revealDeposit(fundingTx, reveal)
+
+        // TODO: Replace mocks with the real implementation
+        relay.getCurrentEpochDifficulty.returns(
+          SingleP2SHDeposit.chainDifficulty
+        )
+        relay.getPrevEpochDifficulty.returns(SingleP2SHDeposit.chainDifficulty)
```

Comment:
See the comment about the relay in `fixture.ts`.

Reply:
5a91ee6fdedf3ce4a3e707ffc0c4c1013eca15ea